### DISCUSSION TITLE
FluentToolTip in Template Column

### DISCUSSION THREAD

**Discussion by ZnowuJa** [microsoft/fluentui-blazor#2140] [Q&A] [answered]
Hi,
I tried to place a tooltip for actions in TemplateColumn:

`        @* Actions *@
        <TemplateColumn Title="Actions" Align="Align.End">

            <FluentButton Id = "ViewId" IconEnd="@(new Icons.Regular.Size20.BookOpen())" OnClick="@(() => View(context))" Disabled="@actionsDisabled" />
            <FluentTooltip Anchor="ViewId" HideTooltipOnCursorLeave="true" Position=TooltipPosition.Left Delay=50>View Asset</FluentTooltip>
            
            <FluentButton Id="EditId" IconEnd="@(new Icons.Regular.Size20.Edit())" OnClick="@(() => EditType(context))" Disabled="@actionsDisabled" />
            <FluentTooltip Anchor="EditId" HideTooltipOnCursorLeave="true" Position=TooltipPosition.Top Delay=50>Edit Asset</FluentTooltip>
            
            <FluentButton Id="DeleteId" IconEnd="@(new Icons.Regular.Size20.Delete())" OnClick="@(() => Delete(context))" Disabled="@actionsDisabled" />
            <FluentTooltip Anchor="DeleteId" HideTooltipOnCursorLeave="true" Position=TooltipPosition.Top Delay=50>Delete Asset</FluentTooltip>
            
            <FluentButton Id="AdddId" IconEnd="@(new Icons.Regular.Size20.Add())" OnClick="@(() => AddToList(context))" Disabled="@actionsDisabled" />
            <FluentTooltip Anchor="AdddId" HideTooltipOnCursorLeave="true" Position=TooltipPosition.Right Delay=50>Add Asset to update</FluentTooltip>

        </TemplateColumn>`
        
It looks like it is working fine, except that tooltip is visible only on buttons in first row of FluentDataGrid. 
Am I doing it correctly or maybe it is not intendent to use it like I did?
Please have a look at attached small MP4 demonstrating that behaviour.

https://github.com/microsoft/fluentui-blazor/assets/70036908/20b76d16-c899-4ffb-8819-6d4fe83e2628

**Comment by vnbaaij** [comment, marked as answer]
You are giving each type of button the same id in every row. Then you are adding the tooltip to that id, so it will always point that same id (which are the buttons in the first row)

To get a 'unique' id per button, you would need to do something like:

```
<TemplateColumn ...>
    @{
        var id = $"view-{context.id}";
    }
    <FluentButton Id="@id" ...</FluentButton>
   <FluentTooltip Anchor="@id" HideTooltipOnCursorLeave="true" Position=TooltipPosition.Left Delay=50>View Asset</FluentTooltip>
        
</TemplateColumn> 
```

Where '{context.id}' is some unique per row value

**Reply by ZnowuJa** [reply]
Well.... 
Again I made mistake forgetting about controls Id...
Thank you!
